server/src/domain/access/access.core.ts

```diff
@@ -96,6 +96,9 @@ export class AccessCore {
       case Permission.ASSET_DOWNLOAD:
         return !!authUser.isAllowDownload && (await this.repository.asset.hasSharedLinkAccess(sharedLinkId, id));
 
+      case Permission.ASSET_UPLOAD:
+        return authUser.isPublicUser && authUser.isAllowUpload;
```

Comment:
Every auth user in this method will be a public user actually, to we don't need this first part.